src/runtime.rs

```diff
@@ -119,10 +130,9 @@ impl std::fmt::Debug for Revision {
 ///
 /// [dyn-cache]: https://docs.rs/dyn-cache
 pub struct Runtime {
-    revision: Revision,
+    rcs: Arc<RwLock<RevisionControlSystem>>,
```

Comment:
Can you make this an `Arc<RevisionController>`? (assuming it's possible to move RwLock inside the controller like I suggested above).

Reply:
This would obscure the code with implicit locks. I consider this dangerous bc it allows to meet race conditions by maintainers very easily.